homeassistant/components/shelly/sensor.py

```diff
@@ -1134,6 +1141,19 @@ class RpcSensor(ShellyRpcAttributeEntity, SensorEntity):
 
     entity_description: RpcSensorDescription
 
+    def __init__(
+        self,
+        coordinator: ShellyRpcCoordinator,
+        key: str,
+        attribute: str,
+        description: RpcSensorDescription,
+    ) -> None:
+        """Initialize sensor."""
+        super().__init__(coordinator, key, attribute, description)
+
+        if callable(description.options_fn):
```

Comment:
We cant move this to `ShellyRpcAttributeEntity` because:
- for `sensor` platform the type of `_attr_options` is `list[str] | None`
- for `select` platform the type of `_attr_options` is `list[str]`